pywr/recorders/_recorders.pyx

```diff
@@ -1126,6 +1126,55 @@ cdef class NumpyArrayStorageRecorder(NumpyArrayAbstractStorageRecorder):
 NumpyArrayStorageRecorder.register()
 
 
+cdef class NumpyArrayNormalisedStorageRecorder(NumpyArrayAbstractStorageRecorder):
+    """Recorder for timeseries information normalised relative to a control curve for a `Storage` node.
+
+    This class stores normalised storage from a specific node for each time-step of a simulation. The normalisation
+    is relative to a `Parameter` which defines a control curve. The data is normalised such that values of 1, 0 and
+    -1 align with full, at control curve and empty volumes respectively. The data is saved internally using a
+     memory view. The data can be accessed through the `data` attribute or `to_dataframe()` method.
+
+    Parameters
+    ----------
+    model : `pywr.core.Model`
+    node : `pywr.core.Node`
+        Node instance to record.
+    parameter : `Parameter`
+        The control curve for normalisation.
+    temporal_agg_func : str or callable (default="mean")
+        Aggregation function used over time when computing a value per scenario. This can be used
+        to return, for example, the median flow over a simulation. For aggregation over scenarios
```

Comment:
This can be used to return, for example, the minimum normalised storage reached over a simulation. For aggregation over scenarios  see the `agg_func` keyword argument.